File: source/dotnet/Library/AdaptiveCards/AdaptiveWarning.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
namespace AdaptiveCards
{
    // TODO: decide if standard error codes are useful or if this should just be string

    /// <summary>
    /// Represents a warning generated while parsing.
    /// </summary>
    public class AdaptiveWarning
    {
        // TODO #2749: temporary warning code for fallback card. Remove when common set of error codes created and integrated.

        /// <summary>
        /// Status code representing the type of warning encountered.
        /// </summary>
        public enum WarningStatusCode
        {
            /// <summary>
            /// The supplied schema version wasn't recognized.
            /// </summary>
            UnsupportedSchemaVersion = 7,

            /// <summary>
            /// The supplied language was not valid.
            /// </summary>
            InvalidLanguage = 12,

            /// <summary>
            /// The maximum number of <see cref="AdaptiveAction"/>s was exceeded.
            /// </summary>
            MaxActionsExceeded = 13,

            /// <summary>
            /// An input was marked as requiring input, but didn't have a label.
            /// </summary>
            EmptyLabelInRequiredInput = 14
        };

        /// <summary>
        /// Initializes an <see cref="AdaptiveWarning"/> instance with the given properties.
        /// </summary>
        /// <param name="code">The <see cref="WarningStatusCode"/> encountered.</param>
        /// <param name="message">The warning message.</param>
        public AdaptiveWarning(int code, string message)
        {
            Code = code;
            Message = message;
        }

        /// <summary>
        /// The <see cref="WarningStatusCode"/>.
        /// </summary>
        public int Code { get; set; }

        /// <summary>
        /// The warning message.
        /// </summary>
        public string Message { get; set; }
    }
}

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/AdaptiveWarning.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package io.adaptivecards.renderer;

public class AdaptiveWarning {
    public static final int UNKNOWN_ELEMENT_TYPE = 1;
    public static final int UNABLE_TO_LOAD_IMAGE = 2;
    public static final int INTERACTIVITY_DISALLOWED = 3;
    public static final int MAX_ACTIONS_EXCEEDED = 4;
    public static final int TOGGLE_MISSING_VALUE = 5;
    public static final int SELECT_SHOW_CARD_ACTION = 6;
    public static final int INVALID_COLUMN_WIDTH_VALUE = 7;
    public static final int EMPTY_LABEL_IN_REQUIRED_INPUT = 8;

    private int code;
    private String message;

    public AdaptiveWarning(int code, String message)
    {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }
}
